src/components/Auth.tsx

```diff
@@ -1,47 +1,115 @@
-import React from "react";
-import * as firebaseui from "firebaseui";
+import React, { Component } from "react";
+import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
 import firebase from "firebase";
+import { signInAction, signOutAction } from "../actions/userActions";
+const { connect } = require("react-redux");
 
-require("dotenv").config();
-
-// Your web app's Firebase configuration
-var firebaseConfig = {
+const config = {
   apiKey: "AIzaSyCvl1CTEcEWYM1681gUWSaawnHAV-PEgWo",
   authDomain: "my-recipes-da233.firebaseapp.com",
   databaseURL: "https://my-recipes-da233.firebaseio.com",
   projectId: "my-recipes-da233",
-  storageBucket: "my-recipes-da233.appspot.com",
-  messagingSenderId: "1062905305210",
   appId: "1:1062905305210:web:e81f2b5d293bedc736c3c3",
   measurementId: "G-BJ3YHDCKZV",
 };
-// Initialize Firebase
-firebase.initializeApp(firebaseConfig);
-firebase.analytics();
-
-var uiConfig = {
-  signInSuccessUrl: "https://alex-aranda.com",
-  signInOptions: [
-    // Leave the lines as is for the providers you want to offer your users.
-    firebase.auth.EmailAuthProvider.PROVIDER_ID,
-  ],
-  // tosUrl and privacyPolicyUrl accept either url string or a callback
-  // function.
-  // Terms of service url/callback.
-  tosUrl: "<your-tos-url>",
-  // Privacy policy url/callback.
-  privacyPolicyUrl: function () {
-    window.location.assign("<your-privacy-policy-url>");
-  },
-};
+firebase.initializeApp(config);
+
+interface authProps {
+  displayName: string;
+  isSignedIn: boolean;
+  signIn: (d: any, e: any, u: any) => void;
+  signOut: () => void;
+}
+
+class Auth extends Component<authProps> {

```

Comment:
You can pass a type for the component's state as the second parameter in your angle brackets.

Reply:
Before working on this it might be good to reassess if you need component state for anything other than an error.